### PR TITLE
Iv 846

### LINKED ISSUE #846: RPCs in OCEX Pallet
- [x] Create an RPC in OCEX pallet to retrieve balance state for recover ( returns ob primitivies -> recovery -> ObRecoveryState )
- [x] Create an RPC for fetching individual balances given main account and asset id

### PR DESCRIPTION
**Pull request by 35359595**
## Describe your changes
RPC endpoints exposed through node to collect offchain or hybrid (onchain + offchain) data for recovery;

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] I removed all Clippy and Formatting Warnings. 
- [x] I added required Copyrights.

### CODE REVIEW

File: pallets/ocex/rpc/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+serde = { version = "1.0.136" }
+parity-scale-codec = { workspace = true, features = ["derive"] }
+jsonrpsee = { workspace = true, features = ["client-core", "server", "macros"] }
+sp-rpc = { workspace = true }
+sp-blockchain = { workspace = true }
+sp-api = { workspace = true }
+sp-runtime = { workspace = true }
+orderbook-primitives = { path = "../../../primitives/orderbook" }
+pallet-ocex-runtime-api = { path = "runtime-api" }
+serde_json = "1.0.96"
```
Comment (serhii-temchenko) [inline pallets/ocex/rpc/Cargo.toml]: Could this be taken from the workspace?

File: pallets/ocex/rpc/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[package]
+name = "pallet-ocex-rpc"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+serde = { version = "1.0.136" }
```
Comment (serhii-temchenko) [inline pallets/ocex/rpc/Cargo.toml]: Could this be taken from the workspace?

File: pallets/ocex/src/lib.rs [hunk 3]
```diff
@@ -1233,6 +1232,70 @@ pub mod pallet {
 				false
 			}
 		}
+
+		pub fn get_ob_recover_state() -> Result<
```
Comment (serhii-temchenko) [inline pallets/ocex/src/lib.rs:1241]: Please add some documentation for the public api.

File: pallets/ocex/src/lib.rs [hunk 4]
```diff
@@ -1233,6 +1232,70 @@ pub mod pallet {
...
+				account_ids,
+				balances,
+				snapshot_id,
+				last_processed_block_number,
+				state_change_id,
+				worker_nonce,
+			))
+		}
+
+		pub fn get_balance(from: T::AccountId, of: AssetId) -> Result<Decimal, DispatchError> {
```
Comment (serhii-temchenko) [inline pallets/ocex/src/lib.rs:1292]: Please add some documentation for the public api.

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

File: pallets/ocex/rpc/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+	#[method(name = "ob_getBalance")]
+	fn get_balance(
+		&self,
+		account_id: AccountId,
+		of: AssetId,
+		at: Option<BlockHash>,
+	) -> RpcResult<String>;
+}
+
+/// A structure that represents the Polkadex Rewards RPC, which allows querying
```
Comment (Gauthamastro) [inline pallets/ocex/rpc/src/lib.rs]: Update doc

**Review by Gauthamastro — COMMENTED** [review]
The structure looks good, the only thing remaining is adding a few unit tests

File: pallets/ocex/rpc/Cargo.toml (hunk 1 above)
Comment (35359595) [reply pallets/ocex/rpc/Cargo.toml]: done

File: pallets/ocex/rpc/Cargo.toml (hunk 2 above)
Comment (35359595) [reply pallets/ocex/rpc/Cargo.toml]: done

File: pallets/ocex/src/lib.rs (hunk 3 above)
Comment (35359595) [reply pallets/ocex/src/lib.rs:1241]: done

File: pallets/ocex/src/lib.rs (hunk 4 above)
Comment (35359595) [reply pallets/ocex/src/lib.rs:1292]: done

File: pallets/ocex/rpc/src/lib.rs (hunk 5 above)
Comment (35359595) [reply pallets/ocex/rpc/src/lib.rs]: done

**Review by serhii-temchenko — APPROVED** [review]

**Review by Gauthamastro — APPROVED** [review]

**Pull request merged**